python/buildPythonInterface.py

```diff
@@ -0,0 +1,60 @@
+#/* +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#   Copyright (c) 2011-2016 The plumed team
+#   (see the PEOPLE file at the root of the distribution for a list of names)
+#
+#   See http://www.plumed.org for more information.
+#
+#   This file is part of plumed, version 2.
+#
+#   plumed is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Lesser General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   plumed is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public License
+#   along with plumed.  If not, see <http://www.gnu.org/licenses/>.
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ */
+#
+# This python routine builds an interface between plumed and python using cython
+#
+from distutils.core import setup
+from distutils.extension import Extension
+from Cython.Build import cythonize
+import numpy
+import subprocess
+import os
+
+# Read makefile.conf and get CC and CXX flags
+f = open('../Makefile.conf', 'r')
+for line in f :
+   if "CC=" in line : os.environ["CC"] = line.replace("CC=","").replace("\n","")
+   if "CXX=" in line : os.environ["CXX"] = line.replace("CXX=","").replace("\n","")
+   if "LDSO=" in line : os.environ["LDSHARED"] = line.replace("LDSO=","").replace("\n","")
+f.close()
+print( "Building interface using CC=" + os.environ["CC"] + " , CXX=" + os.environ["CXX"] + " and LDSHARED=" + os.environ["LDSHARED"] )
+
+plumedversion = subprocess.check_output(['../src/lib/plumed', 'info', '--version']).decode("utf-8")
+print( "Creating interface for plumed version " + plumedversion )
+
+setup(
+  name='plumed',
+  version=plumedversion,
+  description='Python interface to PLUMED',
+  author='Gareth A. Tribello',
+  author_email='plumed-users@googlegroups.com',
+  url='http://www.plumed.org',
+  ext_modules = cythonize([
+                  Extension( name="plumed", 
+                             sources=["plumed.pyx"],
+                             library_dirs=["../src/lib/"],
```

Comment:
@gtribello I think the problem on osx is that you should set library_dirs to `${libdir}` variable (set in `Makefile.con`). On osx typically you have to correctly hardcode the path to the library in order to be able to find it later.

A possible solution would be to create two modules, one for usage "without install" and another for usage "after install", and then install the latter. Or maybe to say that python module can only be used if you full install plumed.

Then, there will be another subtle issue: it is now possible to change the prefix when you do `make install` (e.g. `./configure --prefix=/a && make && make prefix=/b install`). So in principle you should detect if libdir has changed and recompile the library when you install.

Perhaps it is simpler if you just rebuild python.so when using `make install` directly in place (with correct include_dirs and library_dirs)